forge-game/src/main/java/forge/game/ability/effects/RepeatEachEffect.java

```diff
@@ -139,6 +137,36 @@ else if (sa.hasParam("DefinedCards")) {
             }
         }
 
+        if (sa.hasParam("RepeatTypes")) {
+            final String repTypes = sa.getParam("RepeatTypes");
+            final List<String> validTypes = new ArrayList<>();
```

Comment:
A list risks you having duplicates, I don't think it should a type more than once?

Reply:
Is there a better thing to use, or should I just make sure not to add duplicates here?
![image](https://user-images.githubusercontent.com/103371817/199078851-06415169-f139-495d-aba6-7d3cf7590dff.png)

Reply:
That's (one of) the purpose of Sets